2006Scape Server/plugins/plugin/npc/shopkeeper/GeneralStoreDialogue.java

```diff
@@ -0,0 +1,22 @@
+package plugin.npc.shopkeeper;
+
+import com.rs2.game.dialog.Dialogue;
+import com.rs2.game.dialog.DialogueFactory;
+
+
+public final class GeneralStoreDialogue extends Dialogue {
+    @Override
+    public void sendDialogues(DialogueFactory factory) {
+        factory.getPlayer().dialoguePlugin = true;
```

Comment:
I would name the variable here dialogue, not factory, since it's a more descriptive name of what it is.